src/ngraph/op/fused/fake_quantize.cpp

```diff
@@ -0,0 +1,139 @@
+            (input_low_shape.size() == 0 ||
+             (input_low_shape.size() == 1 && input_low_shape.at(0) == data_shape.at(1))),
+            "Input low tensor shape: ",
+            input_low_shape,
+            ", must either be a scalar or a vector of size equal to number of channels.");
+        NODE_VALIDATION_CHECK(
+            this,
+            (input_high_shape.size() == 0 ||
+             (input_high_shape.size() == 1 && input_high_shape.at(0) == data_shape.at(1))),
+            "Input high tensor shape: ",
+            input_high_shape,
+            ", must either be a scalar or a vector of size equal to number of channels.");
+        NODE_VALIDATION_CHECK(
+            this,
+            (output_low_shape.size() == 0 ||
+             (output_low_shape.size() == 1 && output_low_shape.at(0) == data_shape.at(1))),
+            "Output low tensor shape: ",
+            output_low_shape,
+            ", must either be a scalar or a vector of size equal to number of channels.");
+        NODE_VALIDATION_CHECK(
+            this,
+            (output_high_shape.size() == 0 ||
+             (output_high_shape.size() == 1 && output_high_shape.at(0) == data_shape.at(1))),
+            "Output high tensor shape: ",
+            output_high_shape,
+            ", must either be a scalar or a vector of size equal to number of channels.");
+    }
+}
+
+NodeVector op::FakeQuantize::decompose_op() const
+{
+    shared_ptr<Node> data{get_argument(0)};
+    shared_ptr<Node> input_low{get_argument(1)};
+    shared_ptr<Node> input_high{get_argument(2)};
+    shared_ptr<Node> output_low{get_argument(3)};
+    shared_ptr<Node> output_high{get_argument(4)};
+
+    NodeVector broadcasted_nodes =
+        numpy_style_broadcast(NodeVector{data, input_low, input_high, output_low, output_high});
+
+    data = broadcasted_nodes.at(0);
+    input_low = broadcasted_nodes.at(1);
+    input_high = broadcasted_nodes.at(2);
+    output_low = broadcasted_nodes.at(3);
+    output_high = broadcasted_nodes.at(4);
+    shared_ptr<Node> levels_minus_one =
+        Constant::create(element::i32,
+                         data->get_shape(),
+                         vector<size_t>(shape_size(data->get_shape()), m_levels - 1));
+
+    // TODO: arogowiec
+    //   Probably should use Quantize -> Dequantize pattern.
+    //   May use ngraph::builder::ScaledQuantize and ngraph::builder::ScaledDequantize.
+    //   For quantize we probably should change implementation of quantization_util::get_scale
+    //   to take levels value in to consideration.
+
+    // if x <= input_low:
+    //     output = output_low
+    // # round halfway cases away from zero
+    // round((x - input_low) / (input_high - input_low) * (levels-1)) /
+    //     (levels-1) * (output_high - output_low) + output_low
+    // elif x > input_high:
+    //     output = output_high
+    // else:
+    //     # input_low < x <= input_high
+    //
+    throw ngraph_error("Not yet implemented");
```

Comment:
@jbobba  Are we okay with `decompose_op` not being implemented yet?

Reply:
i think so. But we should override 'validate_and_infer_types' to provide shape inference while we figure out the decomposition.

Reply:
I've added the implementation. Please have a look now.